Token contract

joenyzio opened the pull request:
Adding token contract and front end. Still error on transaction and open zeppelin import.

critesjosh commented on packages/hardhat/contracts/Token.sol:
```diff
@@ -0,0 +1,215 @@
+                                uint256 value
+                            );
+
+/*>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> ASSET VALUE TRACKING: TOKEN  <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<*/
```
what are all these comment breaks for?

joenyzio replied:
Oh was going to use them to give more context on main sections throughout the contract. Will remove.

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+import SvgIcon from "@mui/material/SvgIcon";
+import { ethers } from "ethers";
+
+function HomeIcon(props) {
```
Are these being used on the page? I see them commented out below. If we don't use them, we should remove the code.

joenyzio replied:
Planning to add later but I'll remove them from the PR.

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+}
+
+const iconStyle = {
+  // backgroundColor: "#000000", // camel cased
```
Commented out code like this should be deleted if it's not being used and doesn't provide any useful info.

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+  const [contractLink, setContractLink] = useState("");
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+
+  //   const [{ data, error, loading }, getSigner] = useSigner();
```
you can remove this line

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+    }
+  }, [network, contractData]);
+
+  const setToken = async () => {
```
my text editor is telling me that this function is never referenced. Should it be called somewhere?

joenyzio replied:
Looks like this is what I needed for `transferToken`. Fixed the syntax using this code so it works now.

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+    setTokenSymbol(tokenSymbol);
+  };
+
+  const getTokenInitialSupply = async () => {
```
The format in this one is a little weird, you are using await syntax and a Promise callback. I prefer the syntax you use in the `getTokenTotal` function. Both work, I think you should be consistent though.

joenyzio replied:
I like the `getTokenTotal` format but it wasn't working for these. It gives a big number error on the first line before giving the chance to convert in later lines. Would like to switch it back but could use help because I couldn't fix it.

joenyzio replied:
So weird I'll check what I was doing wrong. Updated both functions to use this and they're working now.

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+    setTokenBalanceOf(tokenBalanceOf);
+  };
+
+  const transferTokens = async (e) => {
```
The syntax in this function is incorrect. Check how a transaction is sent in the StorageContract interface again and try to mimic what is done here 

https://github.com/celo-org/celo-progressive-dapp-starter/blob/64a59df17b13a95208950524865ae0f5740c651c/packages/react-app/components/StorageContract.tsx#L53

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+        <Box pt={2}>
+          <Card sx={{ minWidth: 275 }}>
+            <CardContent>
+              {/* <HomeIcon style={iconStyle} color="primary" /> */}
```
Should these be removed?

joenyzio replied:
Might add back later they look nice but will remove for now.

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+              >
+                Submit
+              </Button>
+              {/* <Button
```
Should this be removed?

joenyzio commented on packages/hardhat/contracts/Token.sol:
```diff
@@ -0,0 +1,215 @@
+
+    // Fired when an account authorizes another account to spend tokens on its behalf
+
+    event Approval          
```
From an old smart contract sensei changing to normal along with the functions.

joenyzio commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,489 @@
+    setTokenTotal(convertedTotal);
+  };
+
+  const getBalance = async () => {
```
Same thing happened here as in the other function and is why I changed it. Could use help getting the format like `getTokenTotal`.

joenyzio replied:
Updated with syntax shared on other function

critesjosh commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,390 @@
+    setTokenBalance(converted);
+  };
+
+  const getTokenBalanceOf = async (e) => {
```
This one still has weird await + callback syntax

joenyzio commented on packages/react-app/components/TokenContract.tsx:
```diff
@@ -0,0 +1,390 @@
+
+export function TokenContract({ contractData }) {
+  const { kit, address, network, performActions } = useContractKit();
+  const [tokenInput, setTokenInput] = useInput({ type: "text" });
```
Okay removed

joenyzio commented:
Added mint and burn everything works on my end but failed checks and a few merge conflicts. Really close hopefully it's just small stuff but I'm not sure what the issue is before merging.

joenyzio commented:
Updated token contract. Smart contract works but I updated 3 lines of code - not thousands...definitely doing some wrong on these commits and not sure what's going on.

critesjosh commented:
Looks like you are committing some deployment data files. You can remove all of the files in `packages/hardhat/deployments/alfajores/solcInputs` and the same for the other networks. Can you add files that end up here to the `.gitignore` file as well?

joenyzio commented:
More issues. Quitting on this PR and making edits on a new updated branch.

nestorbonilla commented:
Closing PR for outdated.
